packages/webdriverio/src/scripts/isElementDisplayed.ts

```diff
@@ -144,58 +157,16 @@ export default function isElementDisplayed (element: Element): boolean {
             }
 
             if (nodeIsElement(childNode)) {
-                return elementSubtreeHasNonZeroDimensions(childNode)
+                return elementSubtreeHasNonZeroDimension({
+                    element: childNode,
+                    dimension,
+                })
             }
 
             return false
         })
     }
 
-    function elementOverflowsContainer(element: Element) {
-        const cascadedOverflow = cascadedStylePropertyForElement(element, 'overflow')
-        if (cascadedOverflow !== 'hidden') {
-            return false
-        }
-
-        // FIXME: this needs to take into account the scroll position of the element,
-        // the display modes of it and its ancestors, and the container it overflows.
-        // See Selenium's bot.dom.getOverflowState atom for an exhaustive list of edge cases.
-        return true
-    }
-
-    function isElementSubtreeHiddenByOverflow (element: Element): boolean {
-        if (!element) {
-            return false
-        }
-
-        if (!elementOverflowsContainer(element)) {
-            return false
-        }
-
-        if (!element.childNodes.length) {
-            return false
-        }
-
-        // This element's subtree is hidden by overflow if all child subtrees are as well.
-        return [].every.call(element.childNodes, function (childNode: Element) {
```

Comment:
We lose this functionality by removing this, the purpose of this is to check if any child elements of an element are not displayed. We need to keep this unless I missed something

Reply:
After a careful analysis, it seemed that this test `isElementSubtreeHiddenByOverflow` was redundant with the test `elementSubtreeHasNonZeroDimensions`. 
I'm not sure why those two different tests existed in the first place, but `isElementSubtreeHiddenByOverflow` seemed like an inferior version (not testing for path with > 0 stroke, weird recursive implementation). So I got rid of it.
But I agree this change should be carefully tested on a large code-base before being validated.